### ISSUE TITLE
Overhead benchmark: Why client request time (p95 or mean) without an agent is higher than that with an agent? 

### ISSUE THREAD

**Issue by jiwen624** [open-telemetry/opentelemetry-java-instrumentation#5684]
(I'm not sure if it's a bug so I didn't choose any of the templates)

Hello,

I was looking into the overhead benchmark results and observed that the client request time without an agent is higher than that with an agent. For example: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/gh-pages/benchmark-overhead/results/release/summary.txt#L18-L19   It seems strange to me as from my understanding, the agent would add some overhead to the application (or at least it wouldn't accelerate the app), but from the test results, the average client request time is lower when the application is running with the agent.

I've also run the benchmark by myself on my laptop and on a m5.4xlarge EC2 instance. All my test results show the same behaviour. After skimming through the code, I don't see any difference between these test configurations except the `-javaagent` argument in the "with an agent" runs.

I'm wondering if this is expected, any reason that would make the application runs faster with the agent? Thanks!

**Comment by trask** [comment]
cc @breedx-splk

**Comment by breedx-splk** [comment]
Thanks for pinging me on this, appreciate it.

@jiwen624 I think your hunch is right -- there's nothing about instrumentation that should cause the app to perform better. It should conceptually always be slower because it's doing _more_ work.

I suspect it is similar to something we found in our distro specific testing as well. I think what we discovered is that excluding the JFR recording during the warmup phase caused this behavior. Check out https://github.com/signalfx/splunk-otel-java-overhead-test/pull/3 -- it shows a very similar pattern where the agent runs are slightly faster/less overhead. 

I think the next step is propagate that similar change into upstream to see how we do after. I hope I can sneak that in in the next week or so.
